src/lib/model/Async2SyncModel.js

```diff
@@ -0,0 +1,272 @@
+                };
+                logger.push({ message }).log('requestActionMethod message received');
+            })
+            .wait(config.requestProcessingTimeoutSeconds * 1000);
+    }
+
+
+    /**
+     * 
+     * @param {object} cache - the cache instance used to publish message
+     * @param {object} message  - the message used to trigger deferred job
+     * @param {object} args - args passed to channelNameMethod
+     * @returns {Promise} - the promise which resolves when deferred job is invoked
+     */
+    function triggerDeferredJob({ cache, message, args }) {
+        // input validation, it should throws if any of args is invalid
+        argsValidationMethod(args);
+
+        const cn = channelNameMethod(args);
+        return deferredJob(cache, cn).trigger(message);
+    }
+
+    /**
+     * @name generateKey
+     * @description generates the cache key used to store state machine
+     * @param {object} args - args passed to channelNameMethod
+
+     * @returns {string} - the cache key
+     */
+    function generateKey(args) {
+        // input validation, it should throws if any of args is invalid
+        argsValidationMethod(args);
+
+        return `key-${channelNameMethod(args)}`;
+    }
+
+
+    /**
+     * @name injectHandlersContext
+     * @description injects the config into state machine data, so it will be accessible to on transition notification handlers via `this.handlersContext`
+     * @param {Object} config   - config to be injected into state machine data
+     * @returns {Object}        - the altered specStateMachine
+     */
+    function injectHandlersContext(config) {
+        return {
+            ...specStateMachine,
+            data: {
+                handlersContext: {
+                    modelName,
+                    config: { ...config }, // injects config property
+                    requests: new MojaloopRequests({
+                        logger: config.logger,
+                        peerEndpoint: config.peerEndpoint,
+                        alsEndpoint: config.alsEndpoint,
+                        dfspId: config.dfspId,
+                        tls: config.tls,
+                        jwsSign: config.jwsSign,
+                        jwsSignPutParties: config.jwsSignPutParties,
+                        jwsSigningKey: config.jwsSigningKey,
+                        wso2Auth: config.wso2Auth
+                    })
+                }
+            }
+        };
+    }
+
+
+    /**
+     * @name create
+     * @description creates a new instance of state machine specified in specStateMachine ^
+     * @param {Object} data     - payload data
+     * @param {String} key      - the cache key where state machine will store the payload data after each transition
+     * @param {Object} config   - the additional configuration for transition handlers
+     */
+    async function create(data, key, config) {
+        const spec = injectHandlersContext(config, specStateMachine);
```

Comment:
The function signature doesn't match the above function. Is `specStateMachine` global or scoped somewhere outside this maybe?

Reply:
`specStateMachine` is defined in L48